### PR TITLE
Allow oc new-build to create a binary scratch build

### PR DESCRIPTION
**Pull request by smarterclayton**
Support `oc new-build --binary --name=test` which creates a build that has no From, is binary, and can build anything with a docker image.

### CODE REVIEW

**Comment by smarterclayton** [comment]
@csrwng

File: pkg/generate/app/pipeline.go [hunk 1]
```diff
@@ -137,6 +141,78 @@ func (pb *pipelineBuilder) NewBuildPipeline(from string, resolvedMatch *Componen
 	}, nil
 }
 
+/*
+// NewScratchBuildPipeline creates a pipeline without an input image and with an optional source repository.
```
Comment (csrwng) [inline pkg/generate/app/pipeline.go]: Is this here for doc purposes?

**Comment by csrwng** [comment]
Just one question, otherwise LGTM

File: pkg/generate/app/pipeline.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/generate/app/pipeline.go]: Should have been deleted, fixed now.

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**